hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -115,10 +117,9 @@ protected void processNextRecord(HoodieRecord<? extends HoodieRecordPayload> hoo
   }
 
   @Override
-  protected void processNextDeletedKey(String key) {
-    // TODO : If delete is the only block written and/or records are present in parquet file
-    // TODO : Mark as tombstone (optional.empty()) for data instead of deleting the entry
-    records.remove(key);
+  protected void processNextDeletedKey(HoodieKey hoodieKey) {
```

Comment:
so the core issue was that the merge map had the keys removed, so the compactor could not really remove it from the base file?

Reply:
Yes, that's the core issue. Also, in the RealtimeInputFormat, we need to be able to `skip` records when we see an empty payload.